primitives/src/sentry.rs

```diff
@@ -55,7 +69,37 @@ pub enum Event {
     Close,
 }
 
-#[derive(Serialize, Deserialize)]
+impl Event {
+    pub fn is_click_event(&self) -> bool {
```

Comment:
This looks a lot like JS, we don't need this. The match is perfectly fine of doing this + we still use `match` on most places after checking the `is_` method.

Reply:
We use the `is_` methods when we need to filter an array of `evs` not `match`

Reply:
We use the `is_` methods when we need to filter an array of `evs`. Its much cleaner and better than having `match` littered everywhere.
I am not sure how this is a lot like JS. 
An example of a Rust library that uses `is_` https://lib.rs/crates/enum-methods to check enums.

Reply:
We use the `is_` methods when we need to filter an array of `evs`. Its much cleaner and better than having `match` littered everywhere.
I am not sure how this is a lot like JS. 
An example of a Rust crate that uses `is_` https://lib.rs/crates/enum-methods to check enums, another example is https://crates.io/crates/py_literal

Reply:
We use the `is_` methods when we need to filter an array of `evs`. Its much cleaner and better than having `match` littered everywhere.
I am not sure how this is a lot like JS. 
An example of a Rust crate that uses `is_` https://lib.rs/crates/enum-methods to check enums, another example is https://crates.io/crates/py_literal

Reply:
Also, are you saying it's much better when filtering in multiple places to use `match` than a helper method that is much cleaner, has an appropriate name because "it looks like JS"?

Reply:
If I understand clearly, is it much better when filtering in multiple places to use `match` than a helper method that has an appropriate name?